src/components/hooks/useBookmarks.js

```diff
@@ -335,53 +331,42 @@ export const usePopulateBookmarks = (bookmarks) => {
     (bookmark) => !bookmark?.materialId?.includes("work-of:")
   );
 
-  const relevantWorksByBookmarkId = bookmarks?.map((bookmark) => {
-    const materialTypes = bookmark?.materialType?.split(" / ");
-    const work = workByIdsDataRemovedDuplicates?.find(
-      (w) => w?.workId === bookmark?.workId
-    );
+  const data = useMemo(() => {
+    if (!bookmarks) return [];
 
-    let manifestationWithCorrectMaterialType =
-      work?.manifestations?.mostRelevant.filter((m) =>
-        isMaterialTypesMatch(materialTypes, m?.materialTypes)
+    const relevantWorksByBookmarkId = bookmarks?.map((bookmark) => {
+      const materialTypes = bookmark?.materialType?.split(" / ");
+      const work = workByIdsDataRemovedDuplicates?.find(
+        (w) => w?.workId === bookmark?.workId
       );
 
-    // if bookmarkId is in specificEdition array, then filter the specific edition out
-    const specificEditionBookmark = specificEditions?.find(
-      (se) => se?.bookmarkId === bookmark?.bookmarkId
-    );
-    if (specificEditionBookmark) {
-      const specificManifestation =
-        manifestationWithCorrectMaterialType?.filter(
-          (m) => m?.pid === specificEditionBookmark?.materialId
+      let manifestationWithCorrectMaterialType =
+        work?.manifestations?.mostRelevant.filter((m) =>
+          isMaterialTypesMatch(materialTypes, m?.materialTypes)
         );
-      manifestationWithCorrectMaterialType = specificManifestation;
-    }
-    return {
-      ...work,
-      bookmarkId: bookmark?.bookmarkId,
-      materialId: bookmark?.materialId,
-      pid: specificEditionBookmark ? bookmark?.materialId : undefined,
-      key: bookmark?.key,
-      workId: bookmark?.workId,
-      manifestations: manifestationWithCorrectMaterialType,
-    };
-  });
 
-  const data = useMemo(() => {
-    if (!bookmarks) return [];
-    //cannot group if there are multiple worktypes for same workid in order
-    const groupedByWorkIdAndType = {};
-    relevantWorksByBookmarkId?.forEach((rw) => {
-      rw.manifestations?.forEach((manifestation) => {
-        const key = rw?.key;
-        if (!groupedByWorkIdAndType[key]) {
-          groupedByWorkIdAndType[key] = [];
-        }
-        groupedByWorkIdAndType[key].push(manifestation);
-      });
+      // if materialId is in specificEdition array, then filter the specific edition out
+      const specificEditionBookmark = specificEditions?.find(
+        (se) => se?.materialId === bookmark?.materialId //TODO 2214 use bookmarkId once we are sure, all bookmarks have it.
+        //(se) => se?.bookmarkId === bookmark?.bookmarkId
+      );
+      if (specificEditionBookmark) {
+        const specificManifestation =
+          manifestationWithCorrectMaterialType?.filter(
+            (m) => m?.pid === specificEditionBookmark?.materialId
+          );
+        manifestationWithCorrectMaterialType = specificManifestation;
+      }
```

Comment:
Jeg tror at du kan slippe for at mutere `manifestationsWithCorrectMaterialType`, slippe for at definere `specificEditions`, og slippe for at definere `specificEditionBookmark`, ved at gøre følgende i stedet:

```
const specificManifestations = manifestationsWithCorrectMaterialType
  ?.filter((m) => m?.pid === bookmark?.materialId)
```

og så nede i return:
```
return {
    ...
    manifestations: !isEmpty(specificManifestation) ? specificManifestation : manifestationWithCorrectMaterialTypes,
    ....
}
```

Men du behøber ikke ændre det. Jeg synes bare at for meget mutation er frustrerende :sweat_smile:

Reply:
med en ændring undgå vi også at tage stilling til diskussionen ovenpå :D